filebeat/input/docker/input.go

```diff
@@ -45,17 +47,30 @@ func NewInput(
 ) (input.Input, error) {
 	cfgwarn.Experimental("Docker input is enabled.")
 
+	// Wrap log input with custom docker settings
 	config := defaultConfig
 	if err := cfg.Unpack(&config); err != nil {
 		return nil, errors.Wrap(err, "reading docker input config")
 	}
 
-	// Wrap log input with custom docker settings
-	if len(config.Containers.IDs) == 0 {
+	// Docker input should make sure that no callers should ever pass empty strings as container IDs
+	// Hence we explicitly make sure that we catch such things and print stack traces in the event of
+	// an invocation so that it can be fixed.
+	ids := make([]string, 0)
+	for _, containerID := range config.Containers.IDs {
+		if containerID != "" {
+			ids = append(ids, containerID)
+		} else {
+			logp.Err("Docker container ID can't be empty for Docker input config")
```

Comment:
In what cases is the container id empty? I wonder if we should log it on the error level, meaning it's something the user can fix and then the error will disappear or it's something that just happens from time to time. In this case we better log it on the info level. 

Do we really need 2 log entries or could we have just one?

Reply:
the 2 log entries are just to have something on record for via `err` and dig further in case of a debug scenario. 

we havent really figured out what the scenario is for an empty container. hence the reasoning for adding stack trace to figure out what that place could be.